federation/pkg/kubefed/util/util.go

```diff
@@ -259,15 +261,22 @@ func buildConfigFromSecret(secret *api.Secret, serverAddress string) (*restclien
 // clientset for either the preferred version or the first listed version (if no preference listed)
 // TODO: We need to evaluate the usage of RESTMapper interface to achieve te same functionality
 func GetVersionedClientForRBACOrFail(hostFactory cmdutil.Factory) (client.Interface, error) {
-	discoveryclient, err := hostFactory.DiscoveryClient()
```

Comment:
opened https://github.com/kubernetes/kubernetes/pull/48077 with this impl as an alternative